app/models/recipe.rb

```diff
@@ -1,3 +1,9 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def total_cost
+   ingredients.sum(:cost)
```

Comment:
....Oh! You did make a `total_cost` method! 👍 I didn't see this at first. As long as this is being called in your view, then you're on the right track. Just make sure to always _remove_ what you refactor and rely on the new code instead.